src/main/java/data/ResponseBuilder.java

```diff
@@ -0,0 +1,26 @@
+package data;
+
+import util.UserAuthUtil;
+
+/** Class to build Login/Logout response objects in JSON format */
+public class ResponseBuilder {
+  /**
+   * @param isLoggedIn user login status
+   * @param redirectURL URL to redirect after login/logout
+   * @return JSON string with login/logout URL
+   *         e.g., logged in: { "isLoggedIn":{"/_ah/logout?continue=%2FLogin"}, "isLoggedOut":{""}}
+   *         logged out: { "isLoggedIn":{""}, "isLoggedOut":{"/_ah/login?continue=%2FLogin"}}
+   */
+  public static String toJson(boolean isLoggedIn, String redirectURL) {
+    return "{ \"isLoggedIn\":{\"" + buildLogout(isLoggedIn, redirectURL) + "\"}, " +
+            "\"isLoggedOut\":{\"" + buildLogin(isLoggedIn, redirectURL) + "\"}}";
+  }
+
+  private static String buildLogin(boolean isLoggedIn, String redirectURL) {
+    return isLoggedIn ? "" : UserAuthUtil.getLoginURL(redirectURL);
```

Comment:
Does UserAuthUtil return the attribute name for these URLs? It seems like it would be more clear to set `"loginUrl": getLoginUrl(redirectUrl)` on the false path here (and the true path for buildLogout.